pkg/util/deployment/deployment_test.go

```diff
@@ -388,6 +391,166 @@ func TestGetOldRCs(t *testing.T) {
 	}
 }
 
+func generatePodTemplateSpec(name, nodeName string, annotations, labels map[string]string) api.PodTemplateSpec {
+	return api.PodTemplateSpec{
+		ObjectMeta: api.ObjectMeta{
+			Name:        name,
+			Annotations: annotations,
+			Labels:      labels,
+		},
+		Spec: api.PodSpec{
+			NodeName: nodeName,
+		},
+	}
+}
+
+func TestEqualIgnoreHash(t *testing.T) {
+	tests := []struct {
+		test           string
+		former, latter api.PodTemplateSpec
+		expected       bool
+	}{
+		{
+			"Same spec, same labels",
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{extensions.DefaultDeploymentUniqueLabelKey: "value-1", "something": "else"}),
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{extensions.DefaultDeploymentUniqueLabelKey: "value-1", "something": "else"}),
+			true,
+		},
+		{
+			"Same spec, only pod-template-hash label value is different",
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{extensions.DefaultDeploymentUniqueLabelKey: "value-1", "something": "else"}),
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{extensions.DefaultDeploymentUniqueLabelKey: "value-2", "something": "else"}),
+			true,
+		},
+		{
+			"Same spec, the former doesn't have pod-template-hash label",
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{"something": "else"}),
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{}, map[string]string{extensions.DefaultDeploymentUniqueLabelKey: "value-2", "something": "else"}),
+			true,
+		},
+		{
+			"Different spec, same labels",
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{"former": "value"}, map[string]string{"something": "else"}),
+			generatePodTemplateSpec("foo", "foo-node", map[string]string{"latter": "value"}, map[string]string{"something": "else"}),
+			false,
+		},
+	}
```

Comment:
Add another test to document when the spec is the same, but labels other than + including the PTHash differ.